### PR TITLE
StatusCondition unit tests [11823]

### PR DESCRIPTION
**Pull request by MiguelCompany**
This PR adds empty implementation of StatusCondition and StatusConditionImpl, along with their unit tests

### CODE REVIEW

File: include/fastdds/dds/core/Entity.hpp [hunk 1]
```diff
@@ -124,10 +124,10 @@ class Entity
      * @brief Allows access to the StatusCondition associated with the Entity
      * @return Reference to StatusCondition object
      */
-    RTPS_DllAPI const StatusCondition& get_statuscondition() const
+    RTPS_DllAPI StatusCondition& get_statuscondition() const
     {
         logWarning(CONDITION, "get_statuscondition method not implemented");
-        return status_condition_;
```
Comment (IkerLuengo) [inline include/fastdds/dds/core/Entity.hpp:130]: Why removing the const in the return?

**Review by IkerLuengo — CHANGES_REQUESTED** [review]

File: src/cpp/fastdds/core/condition/StatusConditionImpl.hpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     * @brief Retrieves the list of communication statuses that are taken into account to determine the trigger_value
+     * @return Status set or default status if it has not been set
+     */
+    const StatusMask& get_enabled_statuses() const;
+
+    /**
+     * @brief Set the trigger value of a specific status
+     * @param status The status for which to change the trigger value
+     * @param trigger_value
+     */
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/fastdds/core/condition/StatusConditionImpl.hpp:81]: This was designed with a single status in mind. We could enforce it with an assert when the implementation is done, as I cannot think of a situation where several statuses are triggered at the same time, but I wouldn't add a precondition on the doxydoc, to allow setting several statuses in the future if necessary.

File: include/fastdds/dds/core/Entity.hpp (hunk 1 above)
Comment (MiguelCompany) [reply include/fastdds/dds/core/Entity.hpp:130]: To allow doing `entity.get_statuscondition().set_enabled_statuses(enabled_statuses)`

File: src/cpp/fastdds/core/condition/StatusConditionImpl.hpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+namespace detail {
+
+struct StatusConditionImpl
+{
+    /**
+     * Construct a StatusConditionImpl object.
+     * @param parent @ref StatusCondition creating this object.
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/fastdds/core/condition/StatusConditionImpl.hpp]: Fixed on 4034bdc

File: test/unittest/dds/core/condition/StatusConditionImplTests.cpp [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    EXPECT_EQ(mask_single.to_string(), uut.get_enabled_statuses().to_string());
+    EXPECT_EQ(ReturnCode_t::RETCODE_OK, uut.set_enabled_statuses(mask_none));
+    EXPECT_EQ(mask_none.to_string(), uut.get_enabled_statuses().to_string());
+    EXPECT_EQ(ReturnCode_t::RETCODE_OK, uut.set_enabled_statuses(mask_all));
+    EXPECT_EQ(mask_all.to_string(), uut.get_enabled_statuses().to_string());
+}
+
+TEST(StatusConditionImplTests, notify_trigger)
+{
+    ConditionNotifier notifier;
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply test/unittest/dds/core/condition/StatusConditionImplTests.cpp]: Fixed on 4a5dcdf

**Review by IkerLuengo — APPROVED** [review]
LGTM

**Pull request merged**